### ISSUE TITLE
Fullstack: Integrate Tools and Gen UI from `gen-ui-python`

### ISSUE THREAD

**Issue by JerePlum99** [langchain-ai/open-canvas#113]
### Backend Tools
- Integrate weather and github tools from [Gen UI Python](https://github.com/bracesproul/gen-ui-python) to cleanly show community how to take this and build on it internally. 

### Frontend Tools + UI with dynamic Canvas
- Incorporate example tools for the agent with gen UI streaming and the ability to use the custom canvas

**Comment by bracesproul** [comment]
This is not something we're planning to support in the near future.

**Comment by JerePlum99** [comment]
@bracesproul Is there any good up-to-date example integrating LangGraph and the AI-SDK then? For those of us building internal tools for companies I've struggled to find any good resource to bootstrap a Langgraph example with auth, tools, memory, and the ability to customize the UI easily

**Comment by bracesproul** [comment]
@JerePlum99 yes [this repo](https://github.com/langchain-ai/langchain-nextjs-template/) has a langchain x ai sdk example ([this PR](https://github.com/langchain-ai/langchain-nextjs-template/pull/65) is brining that code up to the latest ai sdk versions)

We're also going to be releasing a react hook for LangGraph usage (streaming, common chat bot use cases with react apps, etc) & LangGraph APIs to natively support generative UI within ~1-2 weeks from now!

**Comment by JerePlum99** [comment]
@bracesproul Sounds great - appreciate the work and excited for what you guys are building!
